### PR TITLE
Create api information

### PR DESCRIPTION
**Pull request by 24eeeee**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/282

Create new controller InfoController

Add GET endpoint GetApiInformation() without Route just [HttpGet].
Add this endpoint to SkippedEndpoints config section, to remove it from TokenValidation cycle.

Return model:
-Current API version (add readonly ApiVersion property to Startup.cs, current version must be the same as front team used right now, check swagger spec)
-Link to the last swagger used by front team right now

### CODE REVIEW

File: src/ProjectService.Data.Provider.MsSql.Ef/ProjectService.Data.Provider.MsSql.Ef.csproj [hunk 1]
```diff
@@ -7,4 +7,8 @@
     <ProjectReference Include="..\ProjectService.Data.Provider\ProjectService.Data.Provider.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Data.Provider.MsSql.Ef/ProjectService.Data.Provider.MsSql.Ef.csproj:13]: Remove

File: src/ProjectService.Data.Provider/ProjectService.Data.Provider.csproj [hunk 2]
```diff
@@ -6,4 +6,8 @@
     <ProjectReference Include="..\ProjectService.Models.Db\ProjectService.Models.Db.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Data.Provider/ProjectService.Data.Provider.csproj:12]: Remove

File: src/ProjectService.Business/ProjectService.Business.csproj [hunk 3]
```diff
@@ -11,4 +11,8 @@
     <ProjectReference Include="..\ProjectService.Models.Db\ProjectService.Models.Db.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Business/ProjectService.Business.csproj:17]: Remove

File: src/ProjectService.Data/ProjectService.Data.csproj [hunk 4]
```diff
@@ -12,4 +12,8 @@
     <ProjectReference Include="..\ProjectService.Models.Db\ProjectService.Models.Db.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Data/ProjectService.Data.csproj:18]: Remove

File: src/ProjectService.Mappers/ProjectService.Mappers.csproj [hunk 5]
```diff
@@ -8,4 +8,8 @@
     <ProjectReference Include="..\ProjectService.Models.Db\ProjectService.Models.Db.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Mappers/ProjectService.Mappers.csproj:14]: Remove

File: src/ProjectService.Models.Broker/ProjectService.Models.Broker.csproj [hunk 6]
```diff
@@ -2,4 +2,8 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Models.Broker/ProjectService.Models.Broker.csproj:8]: Remove

File: src/ProjectService.Models.Db/ProjectService.Models.Db.csproj [hunk 7]
```diff
@@ -2,4 +2,8 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Models.Db/ProjectService.Models.Db.csproj:8]: Remove

File: src/ProjectService.Models.Dto/ProjectService.Models.Dto.csproj [hunk 8]
```diff
@@ -6,4 +6,8 @@
     <ProjectReference Include="..\ProjectService.Models.Db\ProjectService.Models.Db.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Models.Dto/ProjectService.Models.Dto.csproj:12]: Remove

File: src/ProjectService.Validation/ProjectService.Validation.csproj [hunk 9]
```diff
@@ -8,4 +8,8 @@
     <ProjectReference Include="..\ProjectService.Models.Dto\ProjectService.Models.Dto.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Validation/ProjectService.Validation.csproj:14]: Remove

File: src/ProjectService/ProjectService.csproj [hunk 10]
```diff
@@ -20,4 +20,8 @@
     <ProjectReference Include="..\ProjectService.Mappers\ProjectService.Mappers.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService/ProjectService.csproj:26]: Remove

File: src/ProjectService/Startup.cs [hunk 11]
```diff
@@ -72,42 +107,37 @@ public void ConfigureServices(IServiceCollection services)
...
 
             busConfigurator.AddRequestClient<IGetUserDataRequest>(
-                new Uri($"{rabbitMqConfig.BaseUrl}/{rabbitMqConfig.GetUserDataEndpoint}"),
+                new Uri($"{_rabbitMqConfig.BaseUrl}/{_rabbitMqConfig.GetUserDataEndpoint}"),
                 RequestTimeout.After(ms: 100));
 
             busConfigurator.AddRequestClient<IGetDepartmentRequest>(
-                new Uri($"{rabbitMqConfig.BaseUrl}/{rabbitMqConfig.GetDepartmentDataEndpoint}"),
+                new Uri($"{_rabbitMqConfig.BaseUrl}/{_rabbitMqConfig.GetDepartmentDataEndpoint}"),
                 RequestTimeout.After(ms: 100));
```
Comment (KostinP) [inline src/ProjectService/Startup.cs:141]: Remove (Kernel is updated)

File: test/ProjectService.Broker.UnitTests/ProjectService.Broker.UnitTests.csproj [hunk 12]
```diff
@@ -2,4 +2,8 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.UnitTestKernel" Version="1.0.2" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline test/ProjectService.Broker.UnitTests/ProjectService.Broker.UnitTests.csproj:8]: Remove

File: test/ProjectService.Business.UnitTests/ProjectService.Business.UnitTests.csproj [hunk 13]
```diff
@@ -6,4 +6,8 @@
     <ProjectReference Include="..\..\src\ProjectService.Business\ProjectService.Business.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.UnitTestKernel" Version="1.0.2" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline test/ProjectService.Business.UnitTests/ProjectService.Business.UnitTests.csproj:12]: Remove

File: test/ProjectService.Data.UnitTests/ProjectService.Data.UnitTests.csproj [hunk 14]
```diff
@@ -9,4 +9,8 @@
     <ProjectReference Include="..\..\src\ProjectService.Data.Provider.MsSql.Ef\ProjectService.Data.Provider.MsSql.Ef.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.UnitTestKernel" Version="1.0.2" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline test/ProjectService.Data.UnitTests/ProjectService.Data.UnitTests.csproj:15]: Remove

File: test/ProjectService.Mappers.UnitTests/ProjectService.Mappers.UnitTests.csproj [hunk 15]
```diff
@@ -8,4 +8,8 @@
     <ProjectReference Include="..\..\src\ProjectService.Models.Dto\ProjectService.Models.Dto.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.UnitTestKernel" Version="1.0.2" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline test/ProjectService.Mappers.UnitTests/ProjectService.Mappers.UnitTests.csproj:14]: Remove

File: test/ProjectService.Validation.UnitTests/ProjectService.Validation.UnitTests.csproj [hunk 16]
```diff
@@ -6,4 +6,8 @@
     <ProjectReference Include="..\..\src\ProjectService.Validation\ProjectService.Validation.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.UnitTestKernel" Version="1.0.2" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline test/ProjectService.Validation.UnitTests/ProjectService.Validation.UnitTests.csproj:12]: Remove

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/ProjectService.Data.Provider.MsSql.Ef/ProjectService.Data.Provider.MsSql.Ef.csproj (hunk 1 above)
Comment (24eeeee) [reply src/ProjectService.Data.Provider.MsSql.Ef/ProjectService.Data.Provider.MsSql.Ef.csproj:13]: done

File: src/ProjectService.Data.Provider/ProjectService.Data.Provider.csproj (hunk 2 above)
Comment (24eeeee) [reply src/ProjectService.Data.Provider/ProjectService.Data.Provider.csproj:12]: done

File: src/ProjectService.Business/ProjectService.Business.csproj (hunk 3 above)
Comment (24eeeee) [reply src/ProjectService.Business/ProjectService.Business.csproj:17]: done

File: src/ProjectService.Data/ProjectService.Data.csproj (hunk 4 above)
Comment (24eeeee) [reply src/ProjectService.Data/ProjectService.Data.csproj:18]: done

File: src/ProjectService.Mappers/ProjectService.Mappers.csproj (hunk 5 above)
Comment (24eeeee) [reply src/ProjectService.Mappers/ProjectService.Mappers.csproj:14]: done

File: src/ProjectService.Models.Broker/ProjectService.Models.Broker.csproj (hunk 6 above)
Comment (24eeeee) [reply src/ProjectService.Models.Broker/ProjectService.Models.Broker.csproj:8]: done

File: src/ProjectService.Models.Db/ProjectService.Models.Db.csproj (hunk 7 above)
Comment (24eeeee) [reply src/ProjectService.Models.Db/ProjectService.Models.Db.csproj:8]: done

File: src/ProjectService.Models.Dto/ProjectService.Models.Dto.csproj (hunk 8 above)
Comment (24eeeee) [reply src/ProjectService.Models.Dto/ProjectService.Models.Dto.csproj:12]: done

File: src/ProjectService.Validation/ProjectService.Validation.csproj (hunk 9 above)
Comment (24eeeee) [reply src/ProjectService.Validation/ProjectService.Validation.csproj:14]: done

File: src/ProjectService/ProjectService.csproj (hunk 10 above)
Comment (24eeeee) [reply src/ProjectService/ProjectService.csproj:26]: done

File: test/ProjectService.Broker.UnitTests/ProjectService.Broker.UnitTests.csproj (hunk 12 above)
Comment (24eeeee) [reply test/ProjectService.Broker.UnitTests/ProjectService.Broker.UnitTests.csproj:8]: done

File: test/ProjectService.Business.UnitTests/ProjectService.Business.UnitTests.csproj (hunk 13 above)
Comment (24eeeee) [reply test/ProjectService.Business.UnitTests/ProjectService.Business.UnitTests.csproj:12]: done

File: test/ProjectService.Data.UnitTests/ProjectService.Data.UnitTests.csproj (hunk 14 above)
Comment (24eeeee) [reply test/ProjectService.Data.UnitTests/ProjectService.Data.UnitTests.csproj:15]: done

File: test/ProjectService.Mappers.UnitTests/ProjectService.Mappers.UnitTests.csproj (hunk 15 above)
Comment (24eeeee) [reply test/ProjectService.Mappers.UnitTests/ProjectService.Mappers.UnitTests.csproj:14]: done

File: test/ProjectService.Validation.UnitTests/ProjectService.Validation.UnitTests.csproj (hunk 16 above)
Comment (24eeeee) [reply test/ProjectService.Validation.UnitTests/ProjectService.Validation.UnitTests.csproj:12]: done

File: src/ProjectService/Startup.cs [hunk 17]
```diff
@@ -72,42 +107,34 @@ public void ConfigureServices(IServiceCollection services)
...
         }
```
Comment (KostinP) [inline src/ProjectService/Startup.cs:139]: Remove this and 5 other private methods

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/ProjectService/Startup.cs (hunk 11 above)
Comment (24eeeee) [reply src/ProjectService/Startup.cs]: done

File: src/ProjectService.Broker/ProjectService.Broker.csproj [hunk 18]
```diff
@@ -2,4 +2,8 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <PackageReference Update="LT.DigitalOffice.Kernel" Version="1.1.21" />
+  </ItemGroup>
+
```
Comment (KostinP) [inline src/ProjectService.Broker/ProjectService.Broker.csproj:8]: Remove

File: src/ProjectService/Configuration/RabbitMqConfig.cs [hunk 19]
```diff
@@ -1,8 +1,8 @@
-﻿using LT.DigitalOffice.Kernel.Broker;
+﻿using LT.DigitalOffice.Kernel.Configurations;
 
 namespace LT.DigitalOffice.ProjectService.Configuration
 {
-    public class RabbitMqConfig : BaseRabbitMqOptions
+    public class RabbitMqConfig : BaseRabbitMqConfig
     {
         public string GetFileEndpoint { get; set; }
         public string GetUserDataEndpoint { get; set; }
```
Comment (KostinP) [inline src/ProjectService/Configuration/RabbitMqConfig.cs:8]: Remove this file and folder

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/ProjectService.Broker/ProjectService.Broker.csproj (hunk 18 above)
Comment (24eeeee) [reply src/ProjectService.Broker/ProjectService.Broker.csproj:8]: done

File: src/ProjectService/Configuration/RabbitMqConfig.cs (hunk 19 above)
Comment (24eeeee) [reply src/ProjectService/Configuration/RabbitMqConfig.cs:8]: done

File: src/ProjectService/Startup.cs (hunk 17 above)
Comment (24eeeee) [reply src/ProjectService/Startup.cs]: I think I've got a brain damage, but done

**Review by KostinP — APPROVED** [review]

**Review by TIove — APPROVED** [review]

File: src/ProjectService.Models.Dto/Configurations/RabbitMqConfig.cs [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+        public string AddImageEndpoint { get; set; }
+        public string GetFileEndpoint { get; set; }
+        public string GetDepartmentDataEndpoint { get; set; }
+        public string GetPositionEndpoint { get; set; }
+        public string ChangeUserDepartmentEndpoint { get; set; }
+        public string ChangeUserPositionEndpoint { get; set; }
+        public string GetProjectsEndpoint { get; set; }
+        public string GetProjectEndpoint { get; set; }
+        public string SendEmailEndpoint { get; set; }
+        public string GetTokenEndpoint { get; set; }
```
Comment (djino97) [inline src/ProjectService.Models.Dto/Configurations/RabbitMqConfig.cs:19]: Why so much?

Comment (24eeeee) [reply src/ProjectService.Models.Dto/Configurations/RabbitMqConfig.cs]: Fixed

**Review by djino97 — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**